lib/perl/centreon/script/logAnalyserBroker.pm

```diff
@@ -217,139 +217,171 @@ sub parseFile($$) {
-                                             AND `nagios_server`.`id` = `cfg_nagios`.`nagios_server_id` 
-                                             AND `nagios_server`.`ns_activate` = '1' 
-                                             AND `cfg_nagios`.`nagios_activate` = '1' LIMIT 1");
+            FROM `cfg_nagios`, `nagios_server` 
+            WHERE `nagios_server_id` = '$instance' 
+            AND `nagios_server`.`id` = `cfg_nagios`.`nagios_server_id` 
+            AND `nagios_server`.`ns_activate` = '1' 
+            AND `cfg_nagios`.`nagios_activate` = '1' 
+            LIMIT 1");
         if ($status == -1) {
-            die "Can't get information on poller";
+            $self->{logger}->writeLogError("Can't get information on poller");
+            return;
         }
         my $data = $sth->fetchrow_hashref();
         if (!$data->{'log_archive_path'}) {
-            die "Could not find local var log directory";
+            $self->{logger}->writeLogError("Can't find local varlib directory");
+            return;
         }
         $archives = $data->{log_archive_path};
     } else {
         $archives = $self->{centreon_config}->{VarLib} . "/log/$instance/archives/";
     }
 
-    my @log_files = split /\s/,`ls $archives`;
-    foreach (@log_files) {
-        $self->parseFile($archives.$_, $tmp_server->{'name'});
+    opendir (DIR, $archives) or $self->{logger}->writeLogError("Can't find $archives directory") and return;
+    while (my $file = readdir(DIR)) {
+        next if ($file =~ /^\./);
+        $self->parseFile($archives.$file, $tmp_server->{'name'});
     }
+    closedir DIR;
 }
 
 sub run {
     my $self = shift;
 
     $self->SUPER::run();
+    $self->{logger}->writeLogInfo("Starting program...");
 
     if (defined($self->{opt_s})) {
         if ($self->{opt_s} !~ m/\d{2}-\d{2}-\d{4}/) {
             $self->{logger}->writeLogError("Invalid start date provided");
-            exit 1;
+            return;
         }
     }
 
     # Get conf Data
-    my ($status, $sth_config) = $self->{csdb}->query("SELECT `archive_log`, `archive_retention`, `nagios_log_file`  FROM `config` LIMIT 1");
-    die("Cannot get archive log path") if ($status == -1);
+    my ($status, $sth_config) = $self->{csdb}->query("SELECT `archive_retention` FROM `config` LIMIT 1");
+    if ($status == -1) {
+        $self->{logger}->writeLogError("Can't get logs retention duration");
+        return;
+    }
     my $data = $sth_config->fetchrow_hashref();
-    die("Cannot get archive log path") if (!$data->{'archive_log'});
-
-    my $retention = $data->{'archive_retention'};
 
     my ($day,$month,$year) = (localtime(time()))[3,4,5];
-    $self->{retention_time} =  mktime(0,0,0,$day-$retention,$month,$year,0,0,-1);
+    $self->{retention_time} =  mktime(0,0,0,$day-$data->{'archive_retention'},$month,$year,0,0,-1);
 
     # Get cache
     my $filter_instance = "";
-    my $instanceId;
+    my $instance_id;
```

Comment:
Method and variable names must be in lowerCamelCase (from: https://github.com/centreon/centreon/tree/2.8.x/doc/coding-style/php)